src/librustc_platform_intrinsics/lib.rs

```diff
@@ -37,60 +36,41 @@ pub enum IntrinsicDef {
     Named(&'static str),
 }
 
+
+static VOID: Type = Type::Void;
+
 static I8: Type = Type::Integer(true, 8, 8);
 static I16: Type = Type::Integer(true, 16, 16);
 static I32: Type = Type::Integer(true, 32, 32);
 static I64: Type = Type::Integer(true, 64, 64);
-static U8: Type = Type::Integer(false, 8, 8);
-static U16: Type = Type::Integer(false, 16, 16);
-static U32: Type = Type::Integer(false, 32, 32);
-static U64: Type = Type::Integer(false, 64, 64);
```

Comment:
The error is happening with the type `i16x8`. 

As I said previously

This occured in `/rust/src/test/run-pass/simd-upgraded.rs` I have an `i16x8` type defined in `/rust/src/librustc_platform_intrinsics/lib.rs`, but in the error message there isn't anything mentioning i16x8 as a possible size for a 16byte tuple type.